app/test/test_api/test_user.py

```diff
@@ -0,0 +1,143 @@
+    assert response_body['users'] == test_users
+
+
+def test_get_id(app):
+    for i in range(6):
+        response = app.test_client().get('/api/user/{}'.format(i+1))
+        assert response.status_code == 200
+        response_body = response.get_json()
+        assert response_body['user'] == test_users[i]
+
+
+def test_get_id_false(app):
+    response = app.test_client().get('/api/user/9')
+    assert response.status_code == 400
+    response_body = response.get_json()
+    assert response_body['code'] == 400
+    assert response_body['description'] == "None exist user"
+    assert response_body['name'] == "Bad Request"
+
+
+def test_post(app):
+    data_sended = {"name": "zafwfawfzzz", "email": "n.zzfawfawefz.2209@gmail.com"}
+    response = app.test_client().post('/api/user',
+                                      data=json.dumps(data_sended),
+                                      content_type='application/json',)
+
+    assert response.status_code == 200
+    response_body = response.get_json()
+    assert response_body['added_user']['username'] == data_sended['name']
+    assert response_body['added_user']['email'] == data_sended['email']
+
+
+@pytest.mark.parametrize("data_sended, content_type, expected",
+                         [
+                             ("hello", "text/html", "Invalid: content type is not json!"),
+                             ({"gg": "zafwfawfzzz", "email": "n.zzfawfawefz.2209@gmail.com"},
+                              "application/json",
+                              "None username!"),
+                             ({"name": "zafwfawfzzz", "xx": "n.zzfawfawefz.2209@gmail.com"},
+                              "application/json",
+                              "None email!")
+                         ],
+                         )
+def test_post_fail(app, data_sended, content_type, expected):
+    response = app.test_client().post('/api/user',
+                                      data=json.dumps(data_sended),
+                                      content_type=content_type,)
+    assert response.status_code == 400
+    response_body = response.get_json()
+    assert response_body['code'] == 400
+    assert response_body['description'] == expected
+    assert response_body['name'] == "Bad Request"
+
+
+def test_patch(app):
+    data_sended = {"name": "Hiep", "email": "n.vanhiep@gmail.com"}
+    response = app.test_client().patch('/api/user/5',
+                                       data=json.dumps(data_sended),
+                                       content_type='application/json',)
+
+    assert response.status_code == 200
+    response_body = response.get_json()
+    assert response_body['updated_user']['username'] == data_sended['name']
+    assert response_body['updated_user']['email'] == data_sended['email']
+
+
+@pytest.mark.parametrize("data_sended, content_type, expected",
+                         [
+                             ("hello", "text/html", "Invalid: content type is not json!"),
```

Comment:
separate to different variables and insert them later. This is very hard to comprehend what these values mean.